ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,391 @@
+                        message="initremote failed.\nstdout: %s\nstderr: %s" % (e.stdout, e.stderr),
+                        **res_kwargs
+                    )
+                    return
+
+            # 1. create remote object store:
+            # Note: All it actually takes is to trigger the special remote's `prepare` method once.
+            # ATM trying to achieve that by invoking a minimal fsck.
+            # TODO: - It's probably faster to actually talk to the special remote (i.e. pretending to be annex and use the
+            #         protocol to send PREPARE)
+            #       - Alternatively we can create the remote directory and ria version file directly, but this means code
+            #         duplication that then needs to be kept in sync with ria-remote implementation.
+            #       - this leads to the third option: Have that creation routine importable and callable from
+            #         ria-remote package without the need to actually instantiate a RIARemote object
+            lgr.debug("initializing object store")
+            ds.repo.fsck(remote=ria_remote, fast=True, annex_options=['--exclude=*/*'])
+
+        # 2. create a bare repository in-store:
+        # determine layout locations
+        # TODO: This is here for repo_path only. Actually included in target_props['giturl'], but not easily
+        #       accessible.
+        repo_path, archive_path, objects_path = RIARemote.get_layout_locations(base_path, ds.id)
+
+        lgr.debug("init bare repository")
+        # TODO: we should prob. check whether it's there already. How?
+        # Note: like the special remote itself, we assume local FS if no SSH host is specified
+        disabled_hook = repo_path / 'hooks' / 'post-update.sample'
+        enabled_hook = repo_path / 'hooks' / 'post-update'
+
+        if group:
+            chgrp_cmd = "chgrp -R {} {}".format(quote_cmdlinearg(str(group)), quote_cmdlinearg(str(repo_path)))
+
+        if ssh_host:
+            from datalad import ssh_manager
+            ssh = ssh_manager.get_connection(ssh_host, use_remote_annex_bundle=False)
+            ssh.open()
+            ssh('cd {rootdir} && git init --bare{shared}'.format(
+                rootdir=quote_cmdlinearg(str(repo_path)),
+                shared=" --shared='{}'".format(quote_cmdlinearg(shared)) if shared else ''
+            ))
+            if post_update_hook:
+                ssh('mv {} {}'.format(quote_cmdlinearg(str(disabled_hook)),
+                                      quote_cmdlinearg(str(enabled_hook))))
+
+            if group:
+                # Either repository existed before or a new directory was created for it,
+                # set its group to a desired one if was provided with the same chgrp
+                ssh(chgrp_cmd)
+        else:
+            GitRepo(repo_path, create=True, bare=True,
+                    shared=" --shared='{}'".format(quote_cmdlinearg(shared)) if shared else None)
```

Comment:
Quoting is not needed, all valid arguments have no spaces.

Reply:
"Valid arguments" aren't tested anywhere, whenever this is used via python API. And passing through arbitrary raw user input into syscalls because "it will probably somehow fail and that should be fine", doesn't strike me as a desirable direction to go.

Reply:
And yes, that one line doesn't solve that issue - we do it all the time. Still - when and how to start improving on that?